File: android/src/oldarch/com/appodeal/rnappodeal/RNAppodealNativeAdViewManager.java
package com.appodeal.rnappodeal;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.ViewGroupManager;
import com.facebook.react.uimanager.annotations.ReactProp;

import java.util.Map;

public class RNAppodealNativeAdViewManager extends ViewGroupManager<RCTAppodealNativeAdView> {

    private final RNAppodealNativeAdViewManagerImpl impl = new RNAppodealNativeAdViewManagerImpl();

    @NonNull
    @Override
    public String getName() {
        return RNAppodealNativeAdViewManagerImpl.NAME;
    }

    @NonNull
    @Override
    protected RCTAppodealNativeAdView createViewInstance(@NonNull ThemedReactContext reactContext) {
        return impl.createViewInstance(reactContext);
    }

    @Override
    public void onDropViewInstance(@NonNull RCTAppodealNativeAdView view) {
        super.onDropViewInstance(view);
        impl.onDropViewInstance(view);
    }

    @Nullable
    @Override
    public Map<String, Object> getExportedCustomDirectEventTypeConstants() {
        return impl.getExportedCustomDirectEventTypeConstants();
    }

    @ReactProp(name = "adId")
    public void setAdId(RCTAppodealNativeAdView view, @Nullable String value) {
        view.setAdId(value);
    }

    @ReactProp(name = "placement")
    public void setPlacement(RCTAppodealNativeAdView view, @Nullable String value) {
        if (value != null) {
            view.setPlacement(value);
        }
    }
}

File: android/src/newarch/com/appodeal/rnappodeal/RNAppodealNativeAdViewManager.kt
package com.appodeal.rnappodeal

import com.facebook.react.module.annotations.ReactModule
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.ViewGroupManager
import com.facebook.react.uimanager.annotations.ReactProp

@ReactModule(name = RNAppodealNativeAdViewManagerImpl.NAME)
class RNAppodealNativeAdViewManager :
    ViewGroupManager<RCTAppodealNativeAdView>() {

    private val impl = RNAppodealNativeAdViewManagerImpl()

    override fun getName(): String = RNAppodealNativeAdViewManagerImpl.NAME

    override fun createViewInstance(reactContext: ThemedReactContext): RCTAppodealNativeAdView {
        return impl.createViewInstance(reactContext)
    }

    override fun onDropViewInstance(view: RCTAppodealNativeAdView) {
        super.onDropViewInstance(view)
        impl.onDropViewInstance(view)
    }

    override fun getExportedCustomDirectEventTypeConstants(): Map<String?, Any?>? {
        return impl.getExportedCustomDirectEventTypeConstants()
    }

    @ReactProp(name = "adId")
    fun setAdId(view: RCTAppodealNativeAdView, value: String?) {
        view.adId = value
    }

    @ReactProp(name = "placement")
    fun setPlacement(view: RCTAppodealNativeAdView, value: String?) {
        value?.let { view.placement = it }
    }
}
